packages/compass-collection/src/components/collection-header/collection-header.tsx

```diff
@@ -146,102 +82,93 @@ export const CollectionHeader: React.FunctionComponent<
   sourceName,
   editViewName,
   sourcePipeline,
-  stats,
 }) => {
   const darkMode = useDarkMode();
   const showInsights = usePreference('showInsights');
   const {
-    openCollectionsWorkspace,
     openCollectionWorkspace,
     openEditViewWorkspace,
+    openDatabasesWorkspace,
+    openCollectionsWorkspace,
   } = useOpenWorkspace();
 
+  const breadcrumbItems = useMemo(() => {
+    return [
+      {
+        // TODO (COMPASS-7684): add connection name
+        name: 'Cluster',
+        onClick: () => openDatabasesWorkspace(),
+      },
+      {
+        name: toNS(namespace).database,
+        onClick: () => openCollectionsWorkspace(toNS(namespace).database),
+      },
+      // When viewing a view, show the source namespace first
+      sourceName && {
+        name: toNS(sourceName).collection,
+        onClick: () => openCollectionWorkspace(sourceName),
+      },
+      // Show the current namespace
+      {
+        name: toNS(namespace).collection,
+        onClick: () => openCollectionWorkspace(namespace),
+      },
+      // When editing a view, show the view namespace last
+      editViewName && {
+        name: toNS(editViewName).collection,
+        onClick: () => openCollectionWorkspace(editViewName),
+      },
+    ].filter(Boolean) as BreadcrumbItem[];
+  }, [
+    namespace,
+    sourceName,
+    editViewName,
+    openDatabasesWorkspace,
+    openCollectionsWorkspace,
+    openCollectionWorkspace,
+  ]);
+
   const ns = toNS(namespace);
   const database = ns.database;
   const collection = ns.collection;
   const insights =
     showInsights && sourcePipeline?.length
       ? getInsightsForPipeline(sourcePipeline, isAtlas)
       : [];
-
   return (
     <div
+      title={`${database}.${collection}`}
```

Comment:
Nit: you already have a "composed" namespace here
```suggestion
      title={namespace}
```

Reply:
I'd be remove this completely as it adds title to the whole breadcrumb and the badges within this div.